src/libraries/System.Text.Json/tests/Common/CollectionTests/CollectionTests.AsyncEnumerable.cs

```diff
@@ -74,16 +74,24 @@ public async Task WriteNestedAsyncEnumerable_DTO<TElement>(IEnumerable<TElement>
             Assert.Equal(1, asyncEnumerable.TotalDisposedEnumerators);
         }
 
-        [Fact, OuterLoop]
-        public async Task WriteAsyncEnumerable_LongRunningEnumeration_Cancellation()
+        [Theory, OuterLoop]
+        [InlineData(5000, 1000, true)]
```

Comment:
What triggered the change to add different delays?

Reply:
The parameters in line 81 are what trigger the bug in question. Essentially, a delay in the enumerator that exceeds the cancellation token delay and which does not get passed the cancellation token.

Reply:
I've added a standalone regression test in any case.